### PR TITLE
WIP Lively Style Guide

### PR DESCRIPTION
**Pull request by neo-tahi**
WIP

We are self-reviewing ourselves as a means of listing out all the things we need to do, before we leave for the day

— @annapetry @DTrejo

### CODE REVIEW

File: app/views/kss/home/styleguide2.html.erb [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+      <img src="/images/profile-no-image.png" class="user-thumbnail">
+      <span>
+        User Name
+      </span>
+    </a>
+    <a href="#" class="navigation-item">Sample Nav</a>
+  </div>
+
+</body>
+</html>
```
Comment (neo-tahi) [inline app/views/kss/home/styleguide2.html.erb]: we are going to want to move styleguide2..erb.html to something like "styleguide_template.erb", which when compiles to styleguide.erb.html via the rake task.

File: app/views/kss/readme.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    source-page-selector-context=".column">
+</div>
+STEP 3: HYDRATION
+This file is responsible for updating styleguide.html with HTML
+from the .html files in /docs/ux, based on the declared attributes
+This is currently `testing.rb`.
+STEP 4: UPDATING
+This process should be run periodically
+(every 1-2 on an active project, maybe more)
+```
```
Comment (neo-tahi) [inline app/views/kss/readme.md]: This writing from Ryan may or may not be up to date in all ways, we should update it before merging.

File: app/views/layouts/kss/application.html.erb [hunk 3]
```diff
@@ -14,6 +14,7 @@
         <ul>
           <li><%= link_to 'Home', kss.root_path %></li>
           <li><%= link_to 'Styleguide', kss.styleguide_path %></li>
+          <li><%= link_to 'Styleguide New', '/kss/styleguide2' %></li>
```
Comment (neo-tahi) [inline app/views/layouts/kss/application.html.erb]: We should sort out this situation.

The three files we will serve include
- the old KSS-only styleguide? Still useful? get rid of it?
- the template of the new styleguide, before it has app HTML in it
- the new live styleguide

We'll need to choose good URLs for these and link them up in the right spots (e.g. readme, etc)
— @DTrejo

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 Tahi::Application.routes.draw do
   mount Kss::Engine => '/kss' if Rails.env.development?
+  get "kss/styleguide2", to: 'kss/home#styleguide2', as: :styleguide_try
+  get "kss/styleguide3", to: 'kss/home#styleguide3', as: :styleguide_try2
```
Comment (neo-tahi) [inline config/routes.rb]: URL stuff to update
— @DTrejo

**Comment by neo-tahi** [comment]
Did we mean to commit `flow_manager.png`?
— @DTrejo

File: spec/features/populate_styleguide.rb [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+# </div>
+#
+# STEP 3: HYDRATION
+# This file is responsible for updating styleguide.html with HTML
+# from the .html files in /docs/ux, based on the declared attributes
+# This is currently `testing.rb`.
+#
+# STEP 4: UPDATING
+# This process should be run periodically
+# (every 1-2 on an active project, maybe more)
```
Comment (neo-tahi) [inline spec/features/populate_styleguide.rb]: remove this as we've moved it into the app/views/kss readme.

File: spec/features/populate_styleguide.rb [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+      has_css?(".card .card-content")
+      card = all(".card-content").last
+      card.click
+      first(".overlay--fullscreen", visible: true)
+      first(".question-text", visible: true)
+      first('input[name="financial_disclosure.commercially_affiliated"]').set(true)
+      first('input[name="received-funding"]').set(true)
+      page.grab(name)
+    end
+
```
Comment (neo-tahi) [inline spec/features/populate_styleguide.rb]: extra space

File: spec/features/populate_styleguide.rb [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  end
+
+
+  private
+
+  def save_html(filename, selector = "")
+    return unless filename
+
+    first("body div")
+    sleep 2.0
```
Comment (neo-tahi) [inline spec/features/populate_styleguide.rb]: is this needed?

File: testing.rb [hunk 8]
```diff
@@ -0,0 +1,81 @@
+# This is the Styleguide `HYDRATION` File
```
Comment (neo-tahi) [inline testing.rb]: we will want to move this into the kss folder maybe? Or move this code into the rake task. I think the rake task is the best spot.
— @DTrejo

File: testing.rb [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    end
+
+    selection = CGI::unescape_html wrapper_div.to_html
+  end
+
+  # return that snippet of markup
+  return selection
+rescue => e
+  p e.inspect
+  p "Error: Could not open `#{name}` in `#{filename}` with selector `#{selector}`"
```
Comment (neo-tahi) [inline testing.rb]: The name variable is not used, I think we should not use it.
— @DTrejo

File: lib/tasks/styleguide.rake [hunk 10]
```diff
@@ -0,0 +1,13 @@
+namespace :styleguide do
+  desc "Generate Live Styleguide"
+  task :generate => :environment  do
+    system "rspec spec/features/populate_styleguide.rb"
+    system "ruby testing.rb"
+  end
+
```
Comment (annapetry) [inline lib/tasks/styleguide.rake]: Should we have a guard clause ensuring the user has the app installed?  And provide an informative error message if not? Or is that a given?

File: testing.rb [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+end
+
+def init
+  # Open the File
+  @styleguide_path = "app/views/kss/home/styleguide2.html.erb"
+  @populated_styleguide_path = "app/views/kss/home/styleguide3.html.erb"
```
Comment (neo-tahi) [inline testing.rb]: these will be changing as we fix the URLs and names of these.

File: testing.rb [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+  styleguide_html = File.open(@styleguide_path, "r").read
+
+  # loop all the source-page-names and set (or reset) the content
+  # nodes = Nokogiri::HTML styleguide_html
+  nodes = Nokogiri::HTML(styleguide_html) { |config| config.strict }
+  element_nodes = nodes.css("*[source-page-name]")
+  element_nodes.each do |ele|
+    # ele.content = $RESET ? "" : get_content(ele)
```
Comment (neo-tahi) [inline testing.rb]: delete this
— @DTrejo

File: app/views/kss/home/styleguide2.html.erb [hunk 13]
```diff
@@ -0,0 +1,742 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
+<html>
+<head>
+  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+  <style media="screen">
+  /* Hide KSS Generated Code for now */
+  .styleguide-example pre,
+  .styleguide-example .styleguide-description,
+  .styleguide-example h3,
```
Comment (annapetry) [inline app/views/kss/home/styleguide2.html.erb]: Remove KSS stuffs.

File: app/views/kss/home/styleguide2.html.erb [hunk 14]
```diff
@@ -0,0 +1,742 @@
...
+          Task Cards
+        </h2>
+        <div element-name="three" source-page-name="dashboard" source-page-selector=".welcome-message"></div>
+
+      </div>
+
+    </div>
+  </div>
+
+  <div class="navigation" style="left: 0; display: none;">
```
Comment (annapetry) [inline app/views/kss/home/styleguide2.html.erb]: Incorporate this Tahi navigation into the guide.  Move toward a Bootstrap-like scrollspy for styleguide nav.

File: lib/tasks/styleguide.rake [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+  # TODO watch task for while we develop on the styleguide?
+  # This is only for development
+  task :watch => :environment do
+    system('nodemon --exec "ruby testing.rb" -w app/views/kss/home/styleguide2.html.erb testing.rb -V')
```
Comment (annapetry) [inline lib/tasks/styleguide.rake]: COOL.

File: spec/features/populate_styleguide.rb [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+      find(".manuscript-container", visible: true)
+      find(".contributors-link").click
+      find(".contributors.active")
+      page.grab(name)
+    end
+
+    scenario "paper_download_document" do
+      visit '/papers/1/edit'
+      find(".manuscript-container", visible: true)
+      find(".downloads-link").click
```
Comment (annapetry) [inline spec/features/populate_styleguide.rb]: Look into how to check for elements on screen and then interact with them more efficiently.

**Comment by afomi** [comment]
closing in favor of /harvested-styleguide

**Pull request closed without merging**